corehq/ex-submodules/pillowtop/processors/elastic.py

```diff
@@ -68,20 +70,24 @@ def process_change(self, change):
 
         if change.deleted and change.id:
             doc = change.get_document()
+            domain = doc.get('domain') if doc else None
+            if not domain:
+                meta = getattr(change, 'metadata')
```

Comment:
dot-syntax is equivalent and preferred to using `getattr`
```suggestion
                meta = change.metadata
```

Reply:
This is to ensure that if 'metadata' is None or not set it will not error out.

Reply:
To prevent an error, we would need to specify a default option in `getattr` in case the attribute does not exist.

Reply:
Added default value in 2a81942dc96dd45d7db41f309382554f0520ff25 to prevent an error from being raised, which is the reason to use getattr here.

Reply:
I do still feel a bit uncomfortable with the prospect of `domain` maybe being missing - that means when that situation arises, we won't multiplex.

Possible alternatives that come to mind are:
1. We identify specific circumstances where that occurs, and make available an explicit signal that it's okay to not multiplex in this instance (assuming it _is_ okay)
2. Since this is only for deletions, could we propagate the deletion to _all_ indices?  Assuming it noops if the doc isn't present, that seems like it'd be in line with our integrity model.

We may also want to log an exception even if we don't fail hard.

Reply:
I added f84af4aa1c012a826c67bba39c7ce23fce14e20d to at least log an error if we attempt to delete a case from case search without specifying a domain. Based on a slack conversation with Ethan, it seems pretty unlikely that we will find ourselves in a scenario where the domain is None. Currently on production, there aren't any cases with a domain set to None, which means any soft deletion will have a domain available since the postgres doc is still available to query. For hard deletions where the postgres doc is no longer available, we can be fairly confident that the ChangeMeta will be populated with the domain based on [publish_case_deleted](https://github.com/dimagi/commcare-hq/blob/2a81942dc96dd45d7db41f309382554f0520ff25/corehq/form_processor/change_publishers.py#L56-L64). This is because our hard deletion logic only deletes a case if the provided case_id **and** domain match, so if the caller didn't specify a domain, the case deletion would fail, and as stated previously, we don't currently have any cases where a domain is None.